### PR TITLE
S3 Avatar Signed Uploads

### LINKED ISSUE #226: Heroku / Cloud compatible Avatar
Currently, Avatars assume file system access on the host. This is an assumption that won't be true of many PaaS providers.

We'll want to use S3 / Google Cloud Storage / <insert file storage solution here> instead.

### PR DESCRIPTION
**Pull request by AMStrix**
Closes #226.

### What this does
- BE provides signed avatar post for client
- FE uploads avatar to S3 using signed post
- BE removes avatars when appropriate
- BE generates a unique filename via UUID (it was MD5 hashing before, but now we are not uploading the file to BE, just the mimetype)
- `Avatar` model uses `hybrid_property` to limit stored data to file name, extracting filename on `set` and constructing full url on `get`
- adds `boto3` aws sdk dependency to BE

### Testing
- install BE deps
- follow S3 bucket & user directions in `README`
- edit a profile, add a photo, change the photo, delete the photo -- ensure the photo renders on the page
- the bucket should only have one avatar per user, and no avatar if its been deleted

### CODE REVIEW

File: backend/grant/user/models.py [hunk 1]
```diff
@@ -24,10 +26,18 @@ class Avatar(db.Model):
...
     user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
     user = db.relationship("User", back_populates="avatar")
 
+    @hybrid_property
+    def image_url(self):
+        return construct_avatar_url(self._image_url)
+
+    @image_url.setter
+    def image_url(self, image_url):
+        self._image_url = extract_avatar_filename(image_url)
```
Comment (wbobeirne) [inline backend/grant/user/models.py]: 👍 this is great, should make it easy to change hosts

File: frontend/client/utils/s3.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export const uploadSignedPost = (file: Blob, s3Data: any) =>
+  new Promise((res, rej) => {
+    const xhr = new XMLHttpRequest();
```
Comment (wbobeirne) [inline frontend/client/utils/s3.ts]: Any reason you used `XMLHttpRequest` over `fetch` / `axios`? Not a big deal, the API is just a little hard to reason about.

Comment (AMStrix) [reply frontend/client/utils/s3.ts]: Ah, yes. I attempted to use `axios` and found that the grantio auth headers were included even when importing directly from the module. S3 seemed unhappy, so I just went with the example code.

**Comment by AMStrix** [comment]
I figured AWS would be an uncontroversial choice due to its ubiquity and economic profile. Are there some strong feels on this?

**Pull request merged**